spec/computer_player_spec.rb

```diff
@@ -1,26 +1,29 @@
 require 'computer_player'
 
-describe ComputerPlayer do
-
-  before(:each) do
-    @computer_player = ComputerPlayer.new('')
-  end
+describe Computer_player do
+  let(:in_out) { Console_in_out.new(StringIO.new, StringIO.new) }
+  let(:board) { Board.new(Validate.new) }
+  let(:computer_player) { Computer_player.new(in_out) }
 
   it 'Has a default name' do
-    expect(@computer_player.id).to eq('Computer')
+    expect(computer_player.id).to eq('Computer')
   end
 
   it 'Always plays O' do
-    expect(@computer_player.mark).to eq('O')
+    expect(computer_player.mark).to eq('O')
+  end
+
+  it 'Fills an array with free squares' do
+    expect(computer_player.get_free_squares(board)).to eq([1, 2, 3, 4, 5, 6, 7, 8, 9])
   end
 
-  it 'Can return a number under 9' do
-    allow(@computer_player).to receive(:rand).and_return(3)
-    expect(@computer_player.make_move(9)).to eq(3)
+  it 'Can return a number from 0' do
+    allow(computer_player).to receive(:select_free_square).and_return(1)
+    expect(computer_player.make_move(board)).to eq(0)
   end
 
-  it 'Returns a number under 400' do
-    allow(@computer_player).to receive(:rand).and_return(133)
-    expect(@computer_player.make_move(400)).to eq(133)
+  it 'Can return a number up to 8' do
+    allow(computer_player).to receive(:select_free_square).and_return(3)
+    expect(computer_player.make_move(board)).to eq(2)
```

Comment:
Maybe these could consolidate to become - selects an available square on the board

Reply:
I say this because my question is, why is it important that the computer can select a number from 0-8?